### DISCUSSION TITLE
How to check whether Kafka Consumer is up or not ?

### DISCUSSION THREAD

**Discussion by sayoojbk** [edenhill/librdkafka#3776] [Q&A]
Any functions available to check whether a kafka consumer is up or not

**Comment by edenhill** [comment]
See https://github.com/edenhill/librdkafka/wiki/FAQ#is-kafka-up and https://github.com/edenhill/librdkafka/wiki/FAQ#i-want-to-get-an-event-when-brokers-come-up-and-down

**Reply by sayoojbk** [reply]
I went through the article, mentioning Kafka broker being up or down. We can quickly implement an event CB with the ALL_BROKERDOWN implementation that you have done. That works great. But my requirement starts like I need a consumer and a producer to exchange messages. Sometimes, when I produce a msg, I want to check if there is any consumer, no matter which topic it is subscribed to, possibly available to subscribe to the topic my producer is producing.

**Reply by sayoojbk** [reply]
It could be as simple as a function listing out all consumers present and connected to a particular broker.

**Reply by edenhill** [reply]
That's not how Kafka is intended to work. Producers and consumers are decoupled and they should not share any knowledge of one another.  Producing messages is cheap, if there's no consumer to read them then just let your compaction/retention policy remove them from the broker after a certain time or size.

**Reply by sayoojbk** [reply]
Make sense that producer and consumer have to be decoupled. But any possibility from the broker side too if I can check for the status ? Just like we are able to list out the topics using the metadata function that you have implemented similarly anything for consumer ?

**Reply by edenhill** [reply]
I highly recommend you not to go down that path. Focus on the data, not on senders or receivers.
